include/SampleTrack.h

```diff
@@ -109,22 +116,52 @@ class SampleTCOView : public TrackContentObjectView
 
 public slots:
 	void updateSample();
-
-
+    void setSFade();
+    void setCosFade();
+    void setLinearFade();
+    void setManualFade();
 
 protected:
 	void contextMenuEvent( QContextMenuEvent * _cme ) override;
 	void mousePressEvent( QMouseEvent * _me ) override;
 	void mouseReleaseEvent( QMouseEvent * _me ) override;
+	void mouseMoveEvent(QMouseEvent* e) override;
 	void dragEnterEvent( QDragEnterEvent * _dee ) override;
 	void dropEvent( QDropEvent * _de ) override;
 	void mouseDoubleClickEvent( QMouseEvent * ) override;
 	void paintEvent( QPaintEvent * ) override;
+	void enterEvent(QEvent *e) override;
+	void leaveEvent(QEvent *e) override;
 
 
 private:
 	SampleTCO * m_tco;
 	QPixmap m_paintPixmap;
+
+	// Members related to fading
+	bool m_leftCorner = false;
+	bool m_rightCorner = false;
```

Comment:
```suggestion
	bool m_moveLeftCorner = false;
	bool m_moveRightCorner = false;
```

As far as I understood from this first review, those are used on the mouse events for dragging the corners right? Maybe call them `m_moveLeftCorner` to make it consistent with the other members like `m_moveLeftP1` for example, also making their function clearer.

Reply:
Yes they are used for that. I'll rename those.